packages/node/src/trpc.ts

```diff
@@ -0,0 +1,44 @@
+import { getCurrentHub } from '@sentry/core';
+import { normalize } from '@sentry/utils';
+
+interface SentryTrpcMiddlewareOptions {
+  attachRpcInput?: boolean;
+}
+
+interface TrpcMiddlewareArguments<T> {
+  path: string;
+  type: 'query' | 'mutation' | 'subscription';
+  next: () => Promise<T>;
+  rawInput: unknown;
+}
+
+/**
+ * Sentry tRPC middleware that names the handling transaction after the called procedure.
+ *
+ * Use the Sentry tRPC middleware in combination with the Sentry server integration. (e.g. express integration or
+ * Next.js SDK)
+ */
+export async function sentryTrpcMiddleware(options: SentryTrpcMiddlewareOptions) {
+  return async function <T>({ path, type, next, rawInput }: TrpcMiddlewareArguments<T>): Promise<T> {
+    const hub = getCurrentHub();
+    const clientOptions = hub.getClient()?.getOptions();
+    const sentryTransaction = hub.getScope()?.getTransaction();
+
+    if (sentryTransaction) {
+      sentryTransaction.setName(`${path}()`, 'route');
```

Comment:
I just use the path without added `()` when logging myself

Reply:
I asked the team if we have precedence or a convention around transaction names for RPCs. I believe it comes down to preference of how you would like it to look in the product.

Here, for tRPC at least, imo it doesn't make too much sense to show the `METHOD` in the transaction name since it is always POST anyways (correct me if I am wrong ^^) and it would just be noise. The method name is still included in the span details if the server is properly instrumented otherwise.

 Considering the point about noise above, I think we should drop the `()`. It just makes the transaction name more lengthy without any real value. I believe I added it at first because it denoted quite well that something was a method call.

Reply:
We should match OpenTelemetry here, which names their spans as: [`$package.$service/$method`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/rpc.md).

We do this already in our OpenTelemetry transformations across the different SDKs.